src/QtLocationPlugin/QGeoFileTileCacheQGC.h

```diff
@@ -18,20 +18,31 @@ class QGeoFileTileCacheQGC : public QGeoFileTileCache
     static quint32 getMaxDiskCacheSetting();
     static void cacheTile(const QString &type, int x, int y, int z, const QByteArray &image, const QString &format, qulonglong set = UINT64_MAX);
     static void cacheTile(const QString &type, const QString &hash, const QByteArray &image, const QString &format, qulonglong set = UINT64_MAX);
-    static QGCFetchTileTask* createFetchTileTask (const QString &type, int x, int y, int z);
-    static QString getCacheFilename() { return QStringLiteral("qgcMapCache.db"); }
+    static QGCFetchTileTask *createFetchTileTask(const QString &type, int x, int y, int z);
+    static QString getDatabaseFilePath() { return _databaseFilePath; }
+    static QString getCachePath() { return _cachePath; }
```

Comment:
Is this guaranteed to be initialized before it's first used?

Reply:
Actually it won't even get used if I remove _updateDiskFreeSpace as mentioned in the other comment

Reply:
Ok, I'd add a comment or an assertion to make sure a future refactoring does not break this.

Reply:
Got it, made it so that it had to be instantiated first to provide a parameter to MapEngine & added a comment. But also MapEngine will go away soon anyways.

Reply:
Also just FYI the TileCacheWorker is the last major thing to do. Currently it has to service up to 3-4 maps simultaneously while doing offline map downloads, sql queries, amongst other stuff which causes significant issues if you have multiple downloads going at once. So I will turn the map engine into a thread pool worker of some kind

Reply:
Sounds good